### PR TITLE
Add doc for share environments

### PR DESCRIPTION
**Pull request by tei-k**
# Description
https://scalar-labs.atlassian.net/browse/DLT-7061

# Done
Added doc

### CODE REVIEW

File: docs/ShareEnvironment.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# How to share environment with others
+
+This explanation is for multiple person operating in the same environment.
+
+## Get a public/private key-pair from the person who built the environment
+
+:memo: Also need all the tfsatte files when using `local` backend
```
Comment (scalar-boney) [inline docs/ShareEnvironment.md:7]: Small spelling mistake in `tfsatte`.
Expected `tfstate`

File: docs/ShareEnvironment.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+- Fix key_path of `example.tfvars` in your local
+
+  ```
+  public_key_path = "./example_key.pub"
+
+  private_key_path = "./example_key"
+  ```
+
+- Update the output value `private_key_path` and `public_key_path` of tfsate
```
Comment (scalar-boney) [inline docs/ShareEnvironment.md:19]: Small spelling mistake in `tfsatte`.
Expected `tfstate`

**Review by scalar-boney — COMMENTED** [review]
Thank you
I left some comments.

File: docs/ShareEnvironment.md (hunk 1 above)
Comment (tei-k) [reply docs/ShareEnvironment.md]: Thanks! fixed in e6261d1.

File: docs/ShareEnvironment.md (hunk 2 above)
Comment (tei-k) [reply docs/ShareEnvironment.md]: ditto.

**Review by scalar-boney — APPROVED** [review]
LGTM!
Thank you

File: docs/ShareEnvironment.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+- Fix key_path of `example.tfvars` in your local
+
+  ```
+  public_key_path = "./example_key.pub"
+
+  private_key_path = "./example_key"
+  ```
+
+- Update the output value `private_key_path` and `public_key_path` of tfstate
```
Comment (feeblefakie) [inline docs/ShareEnvironment.md:19]:
```suggestion
Then, update tfstate file as follows.
```

File: docs/ShareEnvironment.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+  ```
+  cd example/[aws|azure]/network
```
Comment (feeblefakie) [inline docs/ShareEnvironment.md:22]:
```suggestion
  cd /path/to/network-module
```

File: docs/ShareEnvironment.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  terraform init
+  terraform refresh -var-file=example.tfvars
+  ```
+
+## Update the tfstate of the cassandra, scalardl and monitor modules
```
Comment (feeblefakie) [inline docs/ShareEnvironment.md:27]:
```suggestion
## Update the tfstate of the cassandra, scalardl, and monitor modules
```

File: docs/ShareEnvironment.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+- Update the value `private_key_path` and `public_key_path` of `data.terraform_remote_state.network` in each tfstate
```
Comment (feeblefakie) [inline docs/ShareEnvironment.md:29]:
```suggestion
Update the values of `private_key_path` and `public_key_path` of `data.terraform_remote_state.network` in each tfstate and do as follows.
```

**Review by feeblefakie — COMMENTED** [review]
Left some suggestions on the explanation.

I have one question. 
By doing this, an engineer who deployed an environment first can still have access to the environment?
In other words, is it for sharing or delegating?

**Comment by feeblefakie** [comment]
@tei-k I've given some suggestions but are they correct?

Also,  how about this?
> I have one question.
> By doing this, an engineer who deployed an environment first can still have access to the environment?
> In other words, is it for sharing or delegating?

**Comment by tei-k** [comment]
>By doing this, an engineer who deployed an environment first can still have access to the environment?
In other words, is it for sharing or delegating?

@feeblefakie 
Because can still have access to it, so it's for sharing.

File: docs/ShareEnvironment.md [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+```
+cd /path/to/network
+terraform init
+terraform refresh -var-file=example.tfvars
+```
+
+## Update the tfstate of the cassandra, scalardl, and monitor modules
+
+Update the values of `private_key_path` and `public_key_path` of `data.terraform_remote_state.network` in each tfstate and do as follows.
```
Comment (feeblefakie) [inline docs/ShareEnvironment.md:32]: I'm testing with psim environment but I don't have tfstate file.
Do we need to pull ?

Comment (tei-k) [reply docs/ShareEnvironment.md:32]: The tfstate file in the psim environment is on s3, so you don't need to pull it to local.

Comment (feeblefakie) [reply docs/ShareEnvironment.md:32]: Ok, this line applies only for `local` backend. How about the following commands? They need to be done whether or not local?
Anyways, I think it requires some updates.

Comment (tei-k) [reply docs/ShareEnvironment.md:32]: >They need to be done whether or not local?

Yes, This is necessary regardless of the back-end type. (not only for local)

Comment (feeblefakie) [inline docs/ShareEnvironment.md:32]:
```suggestion
Then, do the following to reflect the above changes to each module.
```

File: docs/ShareEnvironment.md [hunk 8]
```diff
@@ -0,0 +1,38 @@
+# How to share environments with others
+
+Non-testing environments tend to be shared and operated with multiple DevOps engineers.
+This document explains how to do it properly.
+
```
[the remark this answers is not in the document]
Comment (tei-k) [reply docs/ShareEnvironment.md:5]: @feeblefakie looking good ! 👍

**Comment by tei-k** [comment]
@feeblefakie 
Thank you for the many suggestions. fixed in c35cfd2.

File: docs/ShareEnvironment.md [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+Non-testing environments tend to be shared and operated with multiple DevOps engineers.
+This document explains how to do it properly.
+## Why we need the following steps
+
+When we build an environment with scalar-terraform, the paths of a specified key pair are stored in tfstate files. Those paths are expanded to the full paths even if they are specified with relative paths, so the paths can be operator dependent variables unless all the operators agree on the same full paths beforehand.
+With the following steps, you actually update tfstate files based on your environment to make it work for you.
+
+Note that if you update shared tfstate files in Cloud storage based on your environment, others can not use the tfstate files and they have to do the same steps again to make it be able to work for them. So, it is more like a delegation of the operator in such case.
+## Why we need the following steps
```
Comment (feeblefakie) [inline docs/ShareEnvironment.md:11]:
```suggestion
```

File: docs/ShareEnvironment.md [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+
+When we build an environment with scalar-terraform, the paths of a specified key pair are stored in tfstate files. Those paths are expanded to the full paths even if they are specified with relative paths, so the paths can be operator dependent variables unless all the operators agree on the same full paths beforehand.
```
Comment (feeblefakie) [inline docs/ShareEnvironment.md:13]:
```suggestion
```

File: docs/ShareEnvironment.md [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+With the following steps, you actually update tfstate files based on your environment to make it work for you.
```
Comment (feeblefakie) [inline docs/ShareEnvironment.md:14]:
```suggestion
```

File: docs/ShareEnvironment.md [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+Note that if you update shared tfstate files in Cloud storage based on your environment, others can not use the tfstate files and they have to do the same steps again to make it be able to work for them. So, it is more like a delegation of the operator in such case.
```
Comment (feeblefakie) [inline docs/ShareEnvironment.md:15]:
```suggestion
```

**Review by feeblefakie — APPROVED** [review]
LGTM! Thanks!

**Comment by feeblefakie** [comment]
@ymorimo I'll merge it but please take a glance when you get a chance.

**Review by ymorimo — COMMENTED** [review]
LGTM!

**Pull request merged**